optuna/storages/_cached_storage.py

```diff
@@ -441,3 +441,37 @@ def _check_trial_is_updatable(trial: FrozenTrial) -> None:
             raise RuntimeError(
                 "Trial#{} has already finished and can not be updated.".format(trial.number)
             )
+
+    def kill_stale_trials(self) -> List[int]:
+        """Kill stale trials.
+
+        Call :meth:`~optuna.storages.RDBStorage.kill_stale_trials` and take consistency of the
+        cached trial and the trial stored in the backend.
+
+        Returns:
+            List of trial IDs of the killed trials.
+        """
+
+        killed_trial_ids = self._backend.kill_stale_trials()
+
+        with self._lock:
+            for trial_id in killed_trial_ids:
+                if trial_id not in self._trial_id_to_study_id_and_number:
+                    continue
+                study_id, number = self._trial_id_to_study_id_and_number[trial_id]
+                study = self._studies[study_id]
+
+                if trial_id not in study.owned_or_finished_trial_ids:
+                    continue
+
+                cached_trial = study.trials[number]
+                self._check_trial_is_updatable(cached_trial)
+                updates = self._get_updates(trial_id)
+                cached_trial.state = TrialState.FAIL
+                updates.state = TrialState.FAIL
+                updates.datetime_complete = datetime.datetime.now()
+                cached_trial.datetime_complete = datetime.datetime.now()
```

Comment:
Shouldn't we use a shared timestamp between `updates.datetime_complete` and `cached_trial.datetime_complete`?
And I couldn't confirm `datetime_complete` is successfully updated in my environment... 🤔 

```
11|1|3|FAIL|2021-01-17 21:57:55.695808|  # <- staled
12|2|3|FAIL|2021-01-17 21:58:33.143769|2021-01-17 21:58:38.184386  # <- raised the exception in `objective`
```

Reply:
I agree with the idea that `updates.datetime_complete` and `cached_trial.datetime_complete` should be the same. I fixed it.

> And I couldn't confirm datetime_complete is successfully updated in my environment... 🤔

How do you check the `datetime_complete`? I could not realize the staled trials in my environment. It's a great help to share your tests.